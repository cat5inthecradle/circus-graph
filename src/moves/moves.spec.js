const { ApolloServer, gql } = require('apollo-server');
const moves = require('.');
const apparatuses = require('../apparatuses');
const { Move } = require('./Move');

const typeDefs = [
  gql`
    type Query
  `,
  moves.typeDef,
  apparatuses.typeDef,
];

describe('moves', () => {
  const fakeData = [
    {
      guid: 'E7FF6711-57A4-4357-A834-9EC230E73196',
      name: 'Meathook',
      apparatuses: [
        'Trapeze',
        'Lyra',
      ],
    },
    {
      guid: 'D282E8D2-442B-4D94-A3CE-F441E814BF42',
      name: 'Owl',
      apparatuses: [
        'Lyra',
      ],
    },
  ];

  const server = new ApolloServer({
    typeDefs,
    resolvers: [moves.resolvers],
  });

  describe('queries', () => {
    describe('all', () => {
      it('returns all moves', async () => {
        // create a test server to test against, using our production typeDefs,
        // resolvers, and dataSources.
        expect(3).toEqual(3);

        Move.all = jest.fn(() => fakeData);

        // run query against the server and snapshot the output
        const res = await server.executeOperation({
          query: gql`
            query Query {
              moves {
                name
              }
            }
          `,
        });

        // Failing because this isn't a good assertion yet.
        expect(res.data).toEqual(fakeData);
      });
    });
  });
});
