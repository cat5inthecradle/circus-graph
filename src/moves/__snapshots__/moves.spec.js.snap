// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`moves queries all returns all moves 1`] = `
Object {
  "data": Object {
    "moves": Array [
      Object {
        "name": "Meathook",
      },
      Object {
        "name": "Owl",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`moves queries all() returns all moves 1`] = `
Object {
  "data": Object {
    "moves": Array [
      Object {
        "name": "Meathook",
      },
      Object {
        "name": "Owl",
      },
      Object {
        "name": "Man in the Moon",
      },
      Object {
        "name": "Salto Sit",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
